composer/loggers/tensorboard_logger.py

```diff
@@ -0,0 +1,118 @@
+            stored. The full path to this directory will be {log_dir}/{run_name}.
+            If not specified, the :attr:`.State.run_name` will be used.
+        log_dir (str, optional): The path to the directory where all the tensorboard logs
+            will be saved. This is also the value that should be specified when starting
+            a tensorboard server. e.g. `tensorboard --logdir={log_dir}`. If not specified
+            `./tensorboard_logs` will be used.
+        flush_interval (int, optional): How frequently by batch to flush the log to a file.
+            For example, a flush interval of 10 means the log will be flushed to a file
+            every 10 batches. The logs will also be automatically flushed at the start and
+            end of every evaluation phase (`EVENT.EVAL_START` and `EVENT.EVAL_END` ),
+            the end of every epoch (`EVENT.EPOCH_END`), and the end of training
+            (`EVENT.FIT_END`). Default: ``100``.
+        rank_zero_only (bool, optional): Whether to log only on the rank-zero process.
+            Recommended to be true since the rank 0 will have access to most global metrics.
+            A setting of `False` may lead to logging of duplicate values.
+            Default: :attr:`True`.
+    """
+
+    def __init__(self,
+                 run_name: Optional[str] = None,
+                 log_dir: Optional[str] = None,
+                 flush_interval: int = 100,
+                 rank_zero_only: bool = True):
+
+        self.run_name = run_name
+        self.log_dir = log_dir
+        self.flush_interval = flush_interval
+        self.rank_zero_only = rank_zero_only
+        self.writer: SummaryWriter = SummaryWriter()
+
+    def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]):
+        del log_level
+
+        if self.rank_zero_only and dist.get_global_rank() != 0:
+            return
+        for tag, data_point in data.items():
+            if isinstance(data_point, str):  # Will error out with weird caffe2 import error.
+                continue
+            # TODO: handle logging non-(scalars/arrays/tensors/strings)
+            # If a non-(scalars/arrays/tensors/strings) is passed, we skip logging it,
+            # so that we do not crash the job.
+            try:
+                self.writer.add_scalar(tag, data_point, global_step=int(state.timestamp.batch))
+            # Gets raised if data_point is not a tensor, array, scalar, or string.
+            except NotImplementedError:
+                pass
+
+    def init(self, state: State, logger: Logger) -> None:
+        if self.run_name is None:
+            self.run_name = state.run_name
+        if self.log_dir is None:
+            self.log_dir = 'tensorboard_logs'
+        summary_writer_log_dir = Path(self.log_dir) / self.run_name
+        self.writer = SummaryWriter(log_dir=summary_writer_log_dir, filename_suffix=self.run_name)
```

Comment:
Curious why do we need `filename_suffix=self.run_name`?

In addition, is it possible to disable automatic flushing (since we do our own in self._flush)? Not sure if the flush interval could be set to a sentenal like -1, or whether we should just make it really big (like 2^63 - 1 -- assuming it's backed by a 64 bit int in tensorboard).

Mainly concerned if an automatic time-based flush coincides when the artifact logger is writing to the trace file when its copying it, resulting in a corrupted trace.

Reply:
> Curious why do we need `filename_suffix=self.run_name`?

good point. we don't need it. removed.


> In addition, is it possible to disable automatic flushing (since we do our own in self._flush)? Not sure if the flush interval could be set to a sentenal like -1, or whether we should just make it really big (like 2^63 - 1 -- assuming it's backed by a 64 bit int in tensorboard).
> 
> Mainly concerned if an automatic time-based flush coincides when the artifact logger is writing to the trace file when its copying it, resulting in a corrupted trace.

Don't think we can prevent the [first flush](https://github.com/tensorflow/tensorboard/blob/master/tensorboard/summary/writer/event_file_writer.py#L207-L208), but I set `flush_secs` to `365 * 3600 * 24` (number of seconds in year). I don't think we will have any jobs running as long as a year!